Update observability Configmap for new Volsync namespace

prsurve opened the pull request.

Shrivaibavi reviewed (commented):
Please check the config again
https://issues.redhat.com/browse/DFBUGS-1293

Need to remove line 20

Please check line 24 again and remove ~
      - _name_="kube_running_pod_ready",namespace="volsync-system|openshift-operators",pod="volsync.*"

prsurve commented:
> Please check the config again https://issues.redhat.com/browse/DFBUGS-1293
> 
> Need to remove line 20
> 
> Please check line 24 again and remove ~ - _name_="kube_running_pod_ready",namespace="volsync-system|openshift-operators",pod="volsync.*"

If you check configmap provided by Gowtham, it has `~` [jira-comment](https://issues.redhat.com/browse/DFBUGS-1293?focusedId=26385894&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-26385894)

Shrivaibavi commented:
> > Please check the config again https://issues.redhat.com/browse/DFBUGS-1293
> > Need to remove line 20
> > Please check line 24 again and remove ~ - _name_="kube_running_pod_ready",namespace="volsync-system|openshift-operators",pod="volsync.*"
> 
> If you check configmap provided by Gowtham, it has `~` [jira-comment](https://issues.redhat.com/browse/DFBUGS-1293?focusedId=26385894&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-26385894)


Please Check the configmap i added as part of my verification, My last comment.

prsurve commented:
> > > Please check the config again https://issues.redhat.com/browse/DFBUGS-1293
> > > Need to remove line 20
> > > Please check line 24 again and remove ~ - _name_="kube_running_pod_ready",namespace="volsync-system|openshift-operators",pod="volsync.*"
> > 
> > 
> > If you check configmap provided by Gowtham, it has `~` [jira-comment](https://issues.redhat.com/browse/DFBUGS-1293?focusedId=26385894&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-26385894)
> 
> Please Check the configmap i added as part of my verification, My last comment.

what happens if we use `~`

Shrivaibavi commented:
> > > > Please check the config again https://issues.redhat.com/browse/DFBUGS-1293
> > > > Need to remove line 20
> > > > Please check line 24 again and remove ~ - _name_="kube_running_pod_ready",namespace="volsync-system|openshift-operators",pod="volsync.*"
> > > 
> > > 
> > > If you check configmap provided by Gowtham, it has `~` [jira-comment](https://issues.redhat.com/browse/DFBUGS-1293?focusedId=26385894&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-26385894)
> > 
> > 
> > Please Check the configmap i added as part of my verification, My last comment.
> 
> what happens if we use `~`

I updated the DFBUGS-1293 and this PR with my observation, testing and verification. 
Added the config map that worked for me in the BZ also reviewed by Gowtham

You can use any symbol(~, etc..) Please feel free to experiment. 
Make sure to check if having line 20 works for you :-)

prsurve commented:
> > > > > Please check the config again https://issues.redhat.com/browse/DFBUGS-1293
> > > > > Need to remove line 20
> > > > > Please check line 24 again and remove ~ - _name_="kube_running_pod_ready",namespace="volsync-system|openshift-operators",pod="volsync.*"
> > > > 
> > > > 
> > > > If you check configmap provided by Gowtham, it has `~` [jira-comment](https://issues.redhat.com/browse/DFBUGS-1293?focusedId=26385894&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-26385894)
> > > 
> > > 
> > > Please Check the configmap i added as part of my verification, My last comment.
> > 
> > 
> > what happens if we use `~`
> 
> I updated the [DFBUGS-1293](https://issues.redhat.com//browse/DFBUGS-1293) and this PR with my observation, testing and verification. Added the config map that worked for me in the BZ also reviewed by Gowtham
> 
> You can use any symbol(~, etc..) Please feel free to experiment. Make sure to check if having line 20 works for you :-)

I tested with the configmap that i have added to PR and that's working.
Can you add more details  what  you mean `Make sure to check if having line 20 works for you :-)`

Shrivaibavi commented:
Configmap I used to verify the BZ

-------------------------------------------------
```
Configmap

------------------

kind: ConfigMap
apiVersion: v1
metadata:
  name: observability-metrics-custom-allowlist
  namespace: open-cluster-management-observability
data:
  metrics_list.yaml: |
    names:
      - odf_system_health_status
      - odf_system_map
      - odf_system_raw_capacity_total_bytes
      - odf_system_raw_capacity_used_bytes
      - ceph_rbd_mirror_snapshot_sync_bytes
      - ceph_rbd_mirror_snapshot_snapshots
    matches:
      - _name_="csv_succeeded",exported_namespace="openshift-storage",name=~"odf-operator.*"
      - _name_="csv_succeeded",exported_namespace="openshift-dr-system",name=~"odr-cluster-operator.*" 
      - _name_="kube_running_pod_ready",namespace="volsync-system|openshift-operators",pod="volsync.*"
```


If the errors in the UI (DR --> Oevrview page) vanishes and volsync operator recovers from degraded to healthy, we are good to go.

prsurve commented:
> Configmap I used to verify the BZ
> 
> ```
> Configmap
> 
> ------------------
> 
> kind: ConfigMap
> apiVersion: v1
> metadata:
>   name: observability-metrics-custom-allowlist
>   namespace: open-cluster-management-observability
> data:
>   metrics_list.yaml: |
>     names:
>       - odf_system_health_status
>       - odf_system_map
>       - odf_system_raw_capacity_total_bytes
>       - odf_system_raw_capacity_used_bytes
>       - ceph_rbd_mirror_snapshot_sync_bytes
>       - ceph_rbd_mirror_snapshot_snapshots
>     matches:
>       - _name_="csv_succeeded",exported_namespace="openshift-storage",name=~"odf-operator.*"
>       - _name_="csv_succeeded",exported_namespace="openshift-dr-system",name=~"odr-cluster-operator.*" 
>       - _name_="kube_running_pod_ready",namespace="volsync-system|openshift-operators",pod="volsync.*"
> ```
> 
> If the errors in the UI (DR --> Oevrview page) vanishes and volsync operator recovers from degraded to healthy, we are good to go.

Using above configmap here is the result

![image](https://github.com/user-attachments/assets/f177bb10-a541-4ffa-9ea0-e495af2ecbf8)

Shrivaibavi commented:
I dont see ```ceph_rbd_mirror_snapshot_sync_time_sum``` in 4.17 doc https://docs.redhat.com/en/documentation/red_hat_openshift_data_foundation/4.17/html/configuring_openshift_data_foundation_disaster_recovery_for_openshift_workloads/monitoring_disaster_recovery_health#enabling-dr-dashboard-on-hub-cluster_monitor-dr and also in Gowtham's or my config or in doc BZ either https://issues.redhat.com/browse/DFBUGS-1367

We need not have that metric anymore. Hence i suggest to remove it.

prsurve commented:
> I dont see `ceph_rbd_mirror_snapshot_sync_time_sum` in 4.17 doc https://docs.redhat.com/en/documentation/red_hat_openshift_data_foundation/4.17/html/configuring_openshift_data_foundation_disaster_recovery_for_openshift_workloads/monitoring_disaster_recovery_health#enabling-dr-dashboard-on-hub-cluster_monitor-dr and also in Gowtham's or my config or in doc BZ either https://issues.redhat.com/browse/DFBUGS-1367
> 
> Below 2 metrics are for DR dashboard - ceph_rbd_mirror_snapshot_sync_bytes - ceph_rbd_mirror_snapshot_snapshots
> 
> other 4 metrics are for storagesystem dashboard. - odf_system_health_status - odf_system_map - odf_system_raw_capacity_total_bytes - odf_system_raw_capacity_used_bytes
> 
> We need not have old metric anymore`ceph_rbd_mirror_snapshot_sync_time_sum`. Looks like our metrics needed a little cleanup. Hence i suggest to remove it.

That was not added by me Please create a igh ssue to remove that. This PR resolves the volsync operator showing as degraded.

prsurve commented:
> I dont see `ceph_rbd_mirror_snapshot_sync_time_sum` in 4.17 doc https://docs.redhat.com/en/documentation/red_hat_openshift_data_foundation/4.17/html/configuring_openshift_data_foundation_disaster_recovery_for_openshift_workloads/monitoring_disaster_recovery_health#enabling-dr-dashboard-on-hub-cluster_monitor-dr and also in Gowtham's or my config or in doc BZ either https://issues.redhat.com/browse/DFBUGS-1367
> 
> Below 2 metrics are for DR dashboard - ceph_rbd_mirror_snapshot_sync_bytes - ceph_rbd_mirror_snapshot_snapshots
> 
> other 4 metrics are for storagesystem dashboard. - odf_system_health_status - odf_system_map - odf_system_raw_capacity_total_bytes - odf_system_raw_capacity_used_bytes
> 
> We need not have old metric anymore`ceph_rbd_mirror_snapshot_sync_time_sum`. Looks like our metrics needed a little cleanup. Hence i suggest to remove it.

@am-agrawa can you take a look here

am-agrawa commented:
> > I dont see `ceph_rbd_mirror_snapshot_sync_time_sum` in 4.17 doc https://docs.redhat.com/en/documentation/red_hat_openshift_data_foundation/4.17/html/configuring_openshift_data_foundation_disaster_recovery_for_openshift_workloads/monitoring_disaster_recovery_health#enabling-dr-dashboard-on-hub-cluster_monitor-dr and also in Gowtham's or my config or in doc BZ either https://issues.redhat.com/browse/DFBUGS-1367
> > Below 2 metrics are for DR dashboard - ceph_rbd_mirror_snapshot_sync_bytes - ceph_rbd_mirror_snapshot_snapshots
> > other 4 metrics are for storagesystem dashboard. - odf_system_health_status - odf_system_map - odf_system_raw_capacity_total_bytes - odf_system_raw_capacity_used_bytes
> > We need not have old metric anymore`ceph_rbd_mirror_snapshot_sync_time_sum`. Looks like our metrics needed a little cleanup. Hence i suggest to remove it.
> 
> @am-agrawa can you take a look here

I am not aware of any new changes w.r.t metrics. All the metrics that were introduced were tested and documented in 4.14 if I remember correctly. However, let's get a confirmation from @netzzer and @GowthamShanmugam.

GowthamShanmugam commented:
ceph_rbd_mirror_snapshot_sync_time_sum can be removed, i think it is not used for any of the releases.

prsurve commented:
I have created a new PR to remove used metrics

prsurve commented:
/cherry-pick release-4.17
